$OpenBSD: patch-src_Xmobar_hs,v 1.1 2016/10/11 10:24:12 dcoppa Exp $

commit f2d33fa9a2a613fd19c41323f2f11941f8a08e27
Author: michael bishop <cleverca22@gmail.com>
Date:   Wed Sep 14 09:18:44 2016 -0300

tell the user what went wrong in the event of a failure

--- src/Xmobar.hs.orig	Mon Sep  5 17:49:18 2016
+++ src/Xmobar.hs	Tue Oct 11 12:10:30 2016
@@ -106,8 +106,8 @@ startLoop xcfg@(XConf _ _ w _ _ _) sig vs = do
 #endif
     eventLoop tv xcfg [] sig
   where
-    handler thing (SomeException _) =
-      void $ putStrLn ("Thread " ++ thing ++ " failed")
+    handler thing (SomeException e) =
+      void $ putStrLn ("Thread " ++ thing ++ " failed: " ++ (show e))
     -- Reacts on events from X
     eventer signal =
       allocaXEvent $ \e -> do
